### PR TITLE
[MODULAR] Fixes runtime with modular persistence, some QOL additions to chaplain

### LINKED ISSUE #19476: The modular persistence save_data proc is being called on people who cryo, but at that point the player's ckey is no longer linked to the body's brain, causing runtimes
## Testmerges:
#19248 
#19308 
#18110 
#17248 

## Reproduction:

When someone cryos, modular persistence will attempt to save that player's data for NIF purposes I presume.
The issue is that at this point, the player has already left their body and thus their brain is not connected to a ckey anymore.
This results in it spitting out a runtime and presumably also not working.

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Fixes https://github.com/Skyrat-SS13/Skyrat-tg/issues/19476

This is also a bit of a qol update for my previous pr, https://github.com/Skyrat-SS13/Skyrat-tg/pull/19016

Makes it so that a high priest may be chosen from existing chaplain crew members when the previous high priest is cryo'd.

Removed the need for some skyrat edits in chaplain.dm

## How This Contributes To The Skyrat Roleplay Experience

Fixes a bug.

As it is now, the high priest role only becomes available when the initial high priest enters cryosleep. Existing chaplains do not get considered and instead the next one who joins the round will become high priest.

This PR changes it so that if there are any chaplains already on the station who can be considered a valid successor, they will first be selected before handing the torch to the next chaplain to join.

On top of that, if the same sect gets selected as the previous one, the favor will not be reset.

## Proof of Testing

<details>
<summary>The chaplain will be notified if succession occurs</summary>
  
image

</details>

## Changelog

:cl:
fix: fixes runtime with modular persistence when a mob enters cryosleep
qol: chaplains who are already on the station may now be chosen as successor to the high priest if the high priest enters cryosleep, and they will be given full control to change the station religion.
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
Weird two changes to put together, unless modular persistence and chaplains are related in some strange way.

**Comment by vinylspiders** [comment]
> Weird two changes to put together, unless modular persistence and chaplains are related in some strange way.

Yes and no--they are not directly related but their code is executed at the same time (when someone enters cryosleep). When I was working on the modular persistence it got me thinking about the chaplain thing because that was something that was on my list, and something I'd been wanting to do. And this is what happened....

I could separate them into two PR's I spose but the modular persistence fix is so simple I thought it wouldn't be a big deal just lumping them together.

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 1]
```diff
@@ -220,10 +220,12 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/computer/cryopod, 32)
 
 			if(mob_occupant.mind)
 				stored_rank = mob_occupant.mind.assigned_role.title
+				if(isnull(stored_ckey))
+					stored_ckey = mob_occupant.mind.key // if mob does not have a ckey and was placed in cryo by someone else, we can get the key this way
```
Comment (vinylspiders) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm:225]: Marking the changes that pertain to modular persistence -- everything else is for chaplain. This is just some extra insurance to make sure we get the ckey.

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 2]
```diff
@@ -220,10 +220,12 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/computer/cryopod, 32)
...
 
 		var/mob/living/carbon/human/human_occupant = occupant
 		if(human_occupant && human_occupant.mind)
-			human_occupant.save_individual_persistence()
+			human_occupant.save_individual_persistence(stored_ckey)
```
Comment (vinylspiders) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm:229]: This is where the ckey gets passed explicitly for the modular persistence.

File: modular_skyrat/modules/modular_persistence/code/modular_persistence.dm [hunk 3]
```diff
@@ -95,8 +95,8 @@ GLOBAL_LIST_INIT(modular_persistence_ignored_vars, list(
 	return returned_list
 
 /// Saves the held persistence data to where it needs to go.
-/datum/modular_persistence/proc/save_data()
-	var/ckey = replacetext(owner.owner?.ckey || owner.brainmob?.ckey, "@", "")
+/datum/modular_persistence/proc/save_data(var/ckey)
+	ckey = replacetext(ckey || owner.owner?.ckey || owner.brainmob?.ckey, "@", "")
 	if(!owner.owner && !owner.brainmob)
```
Comment (vinylspiders) [inline modular_skyrat/modules/modular_persistence/code/modular_persistence.dm:100]: This is the rest of the modular persistence. So two places total. Just basically added an arg option to explicitly pass a ckey and then did that in the cryopod code.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**